File: src/main/java/sensormodels/store/models/MySQLStoreModel.java
package sensormodels.store.models;

import java.sql.PreparedStatement;
import java.sql.SQLException;

public interface MySQLStoreModel {

  String getTableName();

  String getCreateTableQuery();

  String getInsertIntoTableQuery();

  void fillQueryData(PreparedStatement preparedStmt) throws SQLException;
}

File: src/main/java/sensormodels/store/models/MySqlStoreModel.kt
package sensormodels.store.models

import sensormodels.DatabaseModel
import kotlin.Throws
import java.sql.SQLException
import java.sql.PreparedStatement

interface MySqlStoreModel : DatabaseModel {
    val tableName: String?
    val createTableQuery: String?
    val insertIntoTableQuery: String?

    @Throws(SQLException::class)
    fun fillQueryData(preparedStmt: PreparedStatement?)
}